File: src/main/java/frc/robot/commands/conditionals/SmartElevatorUp.java
package frc.robot.commands.conditionals;

import edu.wpi.first.wpilibj.command.ConditionalCommand;
import frc.robot.Robot;
import frc.robot.subsystems.Elevator;
import frc.robot.commands.elevator.AutomaticElevator;
import frc.robot.commands.elevator.SafeElevatorUp;

/**
 * Elevator is SAFE if: Elevator is currently below the danger line and it wants to go above the danger line
 * Otherwise, elevator is UNSAFE
 */
public class SmartElevatorUp extends ConditionalCommand {

    Elevator.ElevatorPosition targetPosition;

    public SmartElevatorUp(Elevator.ElevatorPosition targetPosition) {
        super(new SafeElevatorUp(targetPosition), new AutomaticElevator(targetPosition));
        this.targetPosition = targetPosition;
    }

    @Override
    public boolean condition() {
        return Robot.elevator.getCurrentHeight() <= Elevator.ElevatorPosition.ROCKET_1_C.encPos &&
                targetPosition.encPos >= Elevator.ElevatorPosition.ROCKET_1_C.encPos;
    }
}

File: src/main/kotlin/ktfrc/robot/commands/conditionals/SmartElevatorUp.kt
package ktfrc.robot.commands.conditionals

import edu.wpi.first.wpilibj.command.ConditionalCommand
import ktfrc.robot.Robot
import ktfrc.robot.commands.elevator.AutomaticElevator
import ktfrc.robot.commands.elevator.SafeElevatorUp
import ktfrc.robot.subsystems.ElevatorPosition

/**
 * Elevator is SAFE if: Elevator is currently below the danger line and it wants to go above the danger line
 * Otherwise, elevator is UNSAFE
 */
class SmartElevatorUp(private val targetPosition: ElevatorPosition):
        ConditionalCommand(SafeElevatorUp(targetPosition), AutomaticElevator(targetPosition)) {

    public override fun condition(): Boolean =
            Robot.elevator.getCurrentHeight() <= ElevatorPosition.ROCKET_1_C.encPos &&
            targetPosition.encPos >= ElevatorPosition.ROCKET_1_C.encPos
}